### PR TITLE
Sctp sendv

### PR DESCRIPTION
**Pull request by hoelscher**
Add support for sctp_sendv. 1 Problem with the list with sock_addrs. With the testcase I have add in sctp_sendv.pkt I get back -1 with 22 invalid Argument. Can you take a look on the last sctp_sendv command  in the testcase, please?

### CODE REVIEW

File: gtests/net/packetdrill/run_system_call.c [hunk 1]
```diff
@@ -3294,6 +3330,207 @@ static int syscall_sctp_recvmsg(struct state *state, struct syscall_spec *syscal
...
+                        if (expr->type == EXPR_SOCKET_ADDRESS_IPV4) {
+				size = sizeof(struct sockaddr_in);
+                                memcpy(expr->value.socket_address_ipv4, addr_ptr, size);
+				addr_ptr = addr_ptr + sizeof(struct sockaddr_in);
+                        } else if (expr->type == EXPR_SOCKET_ADDRESS_IPV6) {
+                                size = sizeof(struct sockaddr_in);
+                                memcpy(expr->value.socket_address_ipv6, addr_ptr, size);
+				addr_ptr = addr_ptr + sizeof(struct sockaddr_in6);
+			}
+		}
```
Comment (tuexen) [inline gtests/net/packetdrill/run_system_call.c]: There are at least two problem with the loop:
1. The first two parameters of memcpy() needs to be swapped.
2. The pointer arithmetic addr_prt += sizeof(...) is wrong. For the arithmetic to be correct, addr_ptr needs to be a char *.

Comment (hoelscher) [inline gtests/net/packetdrill/run_system_call.c]: ok, that problems are fixed, but it doesn't work. The testcase said invalid argument.

File: gtests/net/packetdrill/run_system_call.c [hunk 2]
```diff
@@ -3294,6 +3293,208 @@ static int syscall_sctp_recvmsg(struct state *state, struct syscall_spec *syscal
...
+		int addrlen = expression_list_length(addrs_expr_list);
+		int i = 0;
+		size_t size = 0;
+		char *addr_ptr;
+		for (i = 0; i < addrlen; i++) {
+			expr = get_arg(addrs_expr_list, i, error); 
+			if (expr->type == EXPR_SOCKET_ADDRESS_IPV4) {
+				size += sizeof(struct sockaddr_in);
+			} else if (expr->type == EXPR_SOCKET_ADDRESS_IPV6) {
+				size += sizeof(struct sockaddr_in);
```
Comment (tuexen) [inline gtests/net/packetdrill/run_system_call.c]: Should be
```
size += sizeof(struct sockaddr_in6);
```

**Comment by tuexen** [comment]
Merged.
The problem you saw is fixed by https://github.com/nplab/packetdrill/commit/76dc7af5f8c9c9c63abaff586c5279054d5eedea.

**Pull request closed without merging**